apis/secretversion/v1alpha1/secretversion_types.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// SecretVersionState gives the state of a [SecretVersion][google.cloud.secretmanager.v1.SecretVersion], indicating if it can be accessed.
+type SecretVersionState int32
+
+const (
+	// SecretVersionSTATEUNSPECIFIED represents that SecretVersion state is not specified. This value is unused and invalid.
+	SecretVersionSTATEUNSPECIFIED SecretVersionState = 0
+	// SecretVersionENABLED represents that SecretVersion state  may be accessed.
+	SecretVersionENABLED SecretVersionState = 1
+	// SecretVersionDISABLED represents that SecretVersion state may not be accessed, but the secret data
+	// is still available and can be placed back into the [ENABLED] state.
+	SecretVersionDISABLED SecretVersionState = 2
+	// SecretVersionDESTROYED represents that SecretVersion state is destroyed and the secret data is no longer
+	// stored. A version may not leave this state once entered.
+	SecretVersionDESTROYED SecretVersionState = 3
+)
+
+// SecretVersionStateName is to map integers with states
+var SecretVersionStateName = map[int32]string{
+	0: "STATE_UNSPECIFIED",
+	1: "ENABLED",
+	2: "DISABLED",
+	3: "DESTROYED",
+}
+
+// SecretVersionStateValue is to map states with integers
+var SecretVersionStateValue = map[string]int32{
+	"STATE_UNSPECIFIED": 0,
+	"ENABLED":           1,
+	"DISABLED":          2,
+	"DESTROYED":         3,
+}
+
+// SecretVersionParameters defines parameters for a desired Secret Manager's secret version.
+type SecretVersionParameters struct {
+	// SecretRef refers to the secret object(GCP) created in Kubernetes
+	// Required
+	SecretRef string `json:"secretRef,omitempty"`
```

Comment:
Should this be optional if payload is provided?

Reply:
Sorry, the comment is misleading, `SecretRef` is the name of the secret created within GCP . Basically, the secrets manager expects us to create a `Secret` and then it can have multiple `SecretVersion` . It won't be possible to create a `SecretVersion` without the `Secret` .  End users can choose to create the `Secret` manually and just manage `SecretVersion` through crossplane

Reply:
For referencing K8s secret, I am continuing the work here : https://github.com/crossplane/provider-gcp/pull/317 .

Reply:
Also, I wonder if there's a way to enforce the following validation: it is mandatory to pass in either `Payload` or `KubeSecretRef` . Right now, I have kept them both optional